src/data/proxy/LocalTransientUserProfileStore.java

```diff
@@ -0,0 +1,67 @@
+package data.proxy;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import data.proxy.utils.FunctionalUtils;
+import data.structure.UserProfile;
+
+/**
+ * LocalTransientUserProfileStore manages access to the stored user profiles locally in memory.
+ */
+public class LocalTransientUserProfileStore implements UserProfileStore {
```

Comment:
Descriptive class name but super long. Could we get the same idea across with something like InMemUserProfileStore or just TransientUserProfileStore? I think the local part is implied here.